viewing_party/main.py

```diff
@@ -0,0 +1,176 @@
+#Wave 1 
+
+def create_movie(movie_title, genre, rating):
+    new_movie = None
+    if movie_title and genre and rating:
+        new_movie = {
+            "title":movie_title,        "genre": genre,
+            "rating": rating 
+        }
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    watchlist = user_data["watchlist"] 
+    for movie in watchlist:
+        for movie_info in movie.values():
+            if movie_info == title:
+                user_data["watched"].append(movie)
+                watchlist.remove(movie)
+    return user_data
+
+
+#Wave 2
+
+def get_watched_avg_rating(user_data):
+    sum_rating = 0.0
+    watched_list = user_data["watched"]
+    for movie in watched_list:
+        for movie_val in movie.values():
+            if type(movie_val) == float:
+                movie_rating =  movie_val
+                sum_rating += movie_rating
+    if len(user_data["watched"]) == 0: 
+        return 0.0
+    else:
+        avg_rating = sum_rating/len(user_data["watched"])
+    return avg_rating
+
+def get_most_watched_genre(user_data): 
+    watched_list = user_data["watched"]
+    genre_dict = {}
+    for movie in watched_list:
+        for movie_info, info in movie.items():
+            if movie_info == "genre":
+                movie_genre = info
+                if movie_genre not in genre_dict:
+                    genre_dict[movie_genre] = 1
+                else: 
+                    genre_dict[movie_genre] += 1
+    
+    most_watched_genre = None
+    max_found = None
+    
+    for genre, genre_frequency in genre_dict.items():
+        if max_found is None or genre_frequency > max_found:
+            max_found = genre_frequency
+            most_watched_genre = genre
+    return most_watched_genre
+
+#Wave 3 
+
+def get_unique_watched(user_data):
+    u_unique_watched = {}
+    user_titles = []
+    friends_titles = []
+    user_watched_list = user_data["watched"]
+    for movie_dict in user_watched_list:
+        for movie, movie_title in movie_dict.items():
+            user_titles.append(movie_title)
+    print(user_titles)
```

Comment:
Its generally best to remove print statements like this that aren't required before committing and/or submitting